backend/apps/notifications/rate_limiter.py

```
@@ -0,0 +1,52 @@
+    "webhook": 20,
+    "slack": 20,
+}
+
+class ChannelRateLimiter:
+    """
+    Sliding window rate limiter for (user, channel).
+    Uses Redis / Django cache backend.
+    """
+
+    @classmethod
+    def is_allowed(cls, user_id: int, channel_id: str, max_requests: int = None, window_seconds: int = 60) -> Tuple[bool, int]:
```

Ruff (0.15.21)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)